tiledb/api/c_api/buffer/buffer_api.cc

```diff
@@ -71,8 +76,10 @@ capi_return_t tiledb_buffer_get_data(
   ensure_output_pointer_is_valid(data);
   ensure_output_pointer_is_valid(num_bytes);
 
-  *data = buffer->buffer().data();
-  *num_bytes = buffer->buffer().size();
+  span<const char> span = buffer->buffer();
```

Comment:
This would require making `SerializationBuffer` capable of storing mutable data, which on second thought I don't remember why I didn't initially allow it.

Reply:
After discussion we decided not to make any changes here. We are going to need a `const_cast` somewhere either way, as we accept a `const tiledb_buffer_handle_t*` but return a non-const `void*` in this function.